### ISSUE TITLE
Update vets-website linting and testing rules to WCAG 2.1

### ISSUE THREAD

**Issue by rsmithadhoc** [department-of-veterans-affairs/va.gov-team#43988]
## Description

Currently, the rules used in `vets-website` testing are set to [`wcag2a, wcag2aa`](https://github.com/department-of-veterans-affairs/vets-website/search?q=wcag2aa). Because the standard is [WCAG 2.1](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing), we should update these to [`wcag21a, wcag21aa`](https://github.com/dequelabs/axe-core/blob/a3d5cef612013ffe0db5383a4047508654444d3c/doc/API.md#axe-core-tags).

## Tasks

- [x] Run tests with the updated rules to determine if there are issues to fix.
  - [x] If there are issues to fix, document and communicate them to the appropriate team. Create a follow-up issue to check on this at a later date.
- [x] If not, change to the updated rules.

## Acceptance Criteria

- [ ] WCAG 2.1 rules have been enabled. If not possible, any issues are documented/communicated to their respective team and a follow-up issue to check up on it in the future has been created.

**Comment by rsmithadhoc** [comment]
- Had to troubleshoot issues with my local setup and got those figured out.
- Unit tests looked good with the updated rules.
- E2E tests took a while to run and there were some failures. Unfortunately, that terminal output was too long and got overwritten so I could not document them. I'll have to look at a different way to capture those (output to a file or possibly run it through GHA with the updates since that generates a report).

**Comment by rsmithadhoc** [comment]
- I ran the [E2E suite](https://github.com/department-of-veterans-affairs/vets-website/actions/runs/2805451653) with the new rules and it passed. But I'd still like to confirm that it ran all tests.
- I have a [draft PR](https://github.com/department-of-veterans-affairs/vets-website/pull/21870) up but need to fill out the details there after I'm done testing.

**Comment by rsmithadhoc** [comment]
- [PR out for review.](https://github.com/department-of-veterans-affairs/vets-website/pull/21870)

**Comment by rsmithadhoc** [comment]
- [PR ready for review.](https://github.com/department-of-veterans-affairs/vets-website/pull/21870)

**Comment by rsmithadhoc** [comment]
This has been merged, closing this out.
